Clearer Semantics and Naming for Customized Quantization Range Initialization in Observer

paulshaoyuqiao opened the pull request:
Summary:
In this diff, clearer semantics and namings for are introduced by splitting the original `init_dynamic_qrange` into 2 separate `Optional[int]` types `qmin` and `qmax` to avoid the confusion of the parameters with dynamic quantization.

The `qmin` and `qmax` parameters allow customers to specify their own customary quantization range and enables specific use cases for lower bit quantization.

Test Plan:
To assert the correctness and compatibility of the changes with existing observers, on a devvm, execute the following command to run the unit tests:

`buck test //caffe2/test:quantization -- observer`

Differential Revision: D22948334

raghuramank100 commented on torch/quantization/observer.py:
```diff
@@ -99,7 +98,7 @@ class _ObserverBase(ObserverBase):
     """
     def __init__(self, dtype=torch.quint8, qscheme=torch.per_tensor_affine,
-                 reduce_range=False, initial_dynamic_qrange=None):
+                 reduce_range=False, qmin=None, qmax=None):
```
NIT: Can we call this quant_min and quant_max to be consistent with fake quant?

raghuramank100 commented on torch/quantization/observer.py:
```diff
@@ -116,15 +115,16 @@ def __init__(self, dtype=torch.quint8, qscheme=torch.per_tensor_affine,
+        self.has_customized_qrange = (qmin is not None) and (qmax is not None)
+        if self.has_customized_qrange:
+            self._validate_qmin_qmax(qmin, qmax)
+        self.qmin = qmin
```
Same comment here, request making the attribute names consistent.

raghuramank100 reviewed (approved):
Thanks for making this change, please make the names quant_min and quant_max everywhere for observers.

Commit:
Clearer Semantics and Naming for Customized Quantization Range Initialization in Observer (#42602)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/42602

In this diff, clearer semantics and namings for are introduced by splitting the original `init_dynamic_qrange` into 2 separate `Optional[int]` types `qmin` and `qmax` to avoid the confusion of the parameters with dynamic quantization.

The `qmin` and `qmax` parameters allow customers to specify their own customary quantization range and enables specific use cases for lower bit quantization.

Test Plan:
To assert the correctness and compatibility of the changes with existing observers, on a devvm, execute the following command to run the unit tests:

`buck test //caffe2/test:quantization -- observer`

Reviewed By: vkuzo, raghuramank100

Differential Revision: D22948334

fbshipit-source-id: 8f5ff370ec0a03d4fd5e7ebc51a8612cb443ce2d
